### ISSUE TITLE
Mike wants the Template-Java-Spring modernized

### ISSUE THREAD

**Issue by michaelsteven** [ibm-garage-cloud/planning#765]
**Is your feature request related to a problem? Please describe.**
Things that could be improved with the current Template-Java-Spring.  It currently: 
- Uses spring fox for OpenAPI2 instead of springdoc libraries for OpenAPI3
- Demonstrates localization of GreetingService without using Spring localization libraries (message bundles and  an AcceptHeaderLocaleResolver)
- Does not leverage JavaX validation on request objects, and instead preforms conditional checks.
- Doesn’t implement a rest exception handler using controller advice or return a customized error response to prevent info leakage.
- Health check doesn’t leverage the spring boot actuator
- Maybe show some AOP for trace logging, also implement CommonsRequestLoggingFilter
- Maybe add a Pom file in addition to the build.gradle for shops that don’t want to move to gradle.
- Use functional programming / lamdas when possible

**Which persona(s) does this feature benefit (e.g. developer, SRE, etc)?**
Developer

**What is the impact of not having this feature? What will be the benefit of having this feature?**
In as quantifiable terms as possible describe the value brought by this feature. E.g. this task is performed repeatedly and this feature will reduce the time required by 5 minutes and reduce error each time

**Describe the solution you'd like**
Here is an example github repo where spring boot is implemented in a way that demonstrates all of the items listed above.
https://github.com/michaelsteven/springboot-items-api. 

**Describe alternatives you've considered**
Starting a project with the Template-Java-Spring and manually changing the project

**Additional context**
None.

**Comment by mjperrins** [comment]
Hey Mike this is excellent feature request and if closer to real work spring usage would make sense, I would work with @lsteck and @seansund on agreeing the features, and if they agree spring template should evolve then would love your support on this

**Comment by michaelsteven** [comment]
@mjperrins Yes I will commit some time to improving this template and reach out to them.

**Comment by seansund** [comment]
@michaelsteven Thank you, this is a great feature request. I'll take a look at the repo you linked then we can all discuss the best/most efficient way to move this forward.
